bftengine/src/communication/TlsTCPCommunication.cpp

```diff
@@ -243,6 +243,13 @@ class AsyncTlsConnection : public
     return it->second.isReplica;
   }
 
+  void set_no_delay() {
+    asio::ip::tcp::no_delay option;
+    get_socket().set_option(boost::asio::ip::tcp::no_delay(true));
+    get_socket().get_option(option);
```

Comment:
Is there a reason to think that the option wouldn't get set with set_option, such that we have to check it?

Reply:
I'm fine merging this if it's strictly required that nagle is supported to use. What I mean is that we should actually crash if it fails to set. Seems like that's the intent here so +1.